redis/src/bloom.rs

```diff
@@ -0,0 +1,310 @@
+//! Defines types to use with the bloom filter commands.
+
+#[cfg(feature = "bloom")]
+use crate::{
+    from_redis_value, ErrorKind, FromRedisValue, RedisError, RedisResult, RedisWrite, ToRedisArgs,
+    Value,
+};
+use crate::{Commands, Connection};
+use std::collections::HashMap;
+
+/// scaling options used in bf_reserve and bf_insert commands.
+#[derive(Debug)]
+pub enum ScalingOptions {
+    /// Specifies the desired expansion rate for the filter to be created.
+    ExpansionRate(usize),
+    /// Prevents the filter from creating additional sub-filters if initial capacity is reached.
+    NonScaling,
+}
+
+impl ToRedisArgs for ScalingOptions {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        match *self {
+            ScalingOptions::ExpansionRate(expansion) => {
+                out.write_arg(b"EXPANSION");
+                out.write_arg(format!("{expansion}").as_bytes());
+            }
+            ScalingOptions::NonScaling => {
+                out.write_arg(b"NONSCALING");
+            }
+        }
+    }
+}
+
+/// Options for the `bf_insert` command.
+/// See the [Redis documentation](https://redis.io/docs/latest/commands/bf.insert/) for more information.
+#[derive(Default, Debug)]
+pub struct InsertOptions {
+    capacity: Option<i64>,
+    no_create: Option<bool>,
+    scaling: Option<ScalingOptions>,
+    error_rate: Option<f64>,
+}
+
+impl InsertOptions {
+    ///Indicates that the filter should not be created if it does not already exist.
+    pub fn nocreate(mut self) -> Self {
```

Comment:
for all fields, please use `with_X` prefix. We're not very consistent about this, but trying to get better :)

Reply:
👍